add option to specify a custom search engine

mondbaron opened the pull request.

mondbaron committed:
add option to specify a custom search engine

udenr commented on app/src/main/res/values/strings.xml:
```diff
@@ -359,18 +359,23 @@
     </string-array>
+    <string name="pref_custom_search_engine_title">Custom Search Engine</string>
+    <string name="pref_custom_search_engine_summary">The URL to use as \"Custom Search Engine\". The search string will be inserted at %s.</string>
+    <string name="pref_custom_search_engine_default" translatable="false">https://custom-search-engine.com/?q=%s</string>
```
Can you change it to an appropriate default value, like one of the predefined ones. We don't want users to accidentally send their queries to `custom-search-engine.com` if they forget to enter something.

udenr commented on app/src/main/java/com/secuso/privacyfriendlycodescanner/qrscanner/util/WebSearchUtil.kt:
```diff
@@ -70,11 +70,29 @@ object WebSearchUtil {
             context.resources.getStringArray(R.array.pref_search_engine_uris)
-        return searchEngineUris[searchEngineIndex]
+        val pref = PreferenceManager.getDefaultSharedPreferences(context)
+        val customSearchEngineUri = pref.getString("pref_custom_search_engine_uri", searchEngineUris[0])
```
Can you define the key in PreferenceKeys.kt and only use a reference here? This is to prevent typos if used in different parts of the code.
Also add the key to [BackupRestorer.kt](https://github.com/SecUSo/privacy-friendly-qr-scanner/blob/656c29f5374770228c667b23ffa9218c425e93f4/app/src/main/java/com/secuso/privacyfriendlycodescanner/qrscanner/backup/BackupRestorer.kt#L107) to ensure it gets restored correctly.

udenr reviewed (changes requested):
Thank you for this pull request. Can you address a few small changes?

Commit:
Changes default custom search engine to a sane value.
Also refactors new preference key into PreferenceKeys.kt and adds it to the BackupRestorer.kt
